### PR TITLE
Give system key store providers precedence over instance-level providers

### PR DESCRIPTION
**Pull request by johnnypham**
System providers will be checked before command and connection providers.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs [hunk 1]
```diff
@@ -2242,6 +2242,31 @@ public void TestCommandCustomKeyStoreProviderDuringAeQuery(string connectionStri
             }
         }
 
+        // On Windows, "_fixture" will be type SQLSetupStrategyCertStoreProvider
+        // On non-Windows, "_fixture" will be type SQLSetupStrategyAzureKeyVault
+        // Test will pass on both but only SQLSetupStrategyCertStoreProvider is a system provider
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(AEConnectionStringProvider))]
+        public void TestSystemProvidersHavePrecedenceOverInstanceLevelProviders(string connectionString)
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs:2250]: Can we have 1 more test for Connection instance registration similarly?

**Comment by Wraith2** [comment]
Why? this will mean that there is no way to override a system provider, is that the intention?

**Comment by cheenamalhotra** [comment]
System providers are non-overridable by design, even with current static registration API.

**Comment by Wraith2** [comment]
Ok.

**Review by cheenamalhotra — APPROVED** [review]

**Review by DavoudEshtehari — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Pull request merged**